### PR TITLE
Script for submit task (CLI)

### LINKED ISSUE #122: Create SAPS Cli for submit tasks
This function will be something geared towards the administrator who wants to test through a CLI instead of only through the Dashboard. We opted to work behind the Dispatcher backend communication, instead of adding directly to the Catalog. At this point, you want to create a CLI for submitting tasks.

### PR DESCRIPTION
**Pull request by thiagoyeds**
#122 resolved

### CODE REVIEW

File: bin/submit-task [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+## This script submits new tasks for SAPS interacting with Dispatcher backend.
+
+## Checking args
+if [ $# -ne 12 ]
+then
+  echo "Usage: $0 USER_EMAIL USER_PASSWORD LOWER_LEFT_LATITUDE UPPER_RIGHT_LATITUDE LOWER_LEFT_LONGITUDE UPPER_RIGHT_LONGITUDE PROCESSING_INIT_DATE PROCESSING_FINAL_DATE INPUTDOWNLOADING_TAG PREPROCESSING_TAG PROCESSING_TAG DISPATCHER_ACCESS_IP"
+  echo "e.g. $0 <user-email> <user-paswword> -7.913 -6.547 -37.814 -35.757 2015-06-23 2015-06-23 googleapis default ufcg-sebal <dispatcher-access-ip>"
```
Comment (thiagomanel) [inline bin/submit-task]: remove echo from line 9

File: bin/submit-task [hunk 2]
```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+## This script submits new tasks for SAPS interacting with Dispatcher backend.
+
+## Checking args
+if [ $# -ne 12 ]
+then
+  echo "Usage: $0 USER_EMAIL USER_PASSWORD LOWER_LEFT_LATITUDE UPPER_RIGHT_LATITUDE LOWER_LEFT_LONGITUDE UPPER_RIGHT_LONGITUDE PROCESSING_INIT_DATE PROCESSING_FINAL_DATE INPUTDOWNLOADING_TAG PREPROCESSING_TAG PROCESSING_TAG DISPATCHER_ACCESS_IP"
```
Comment (thiagomanel) [inline bin/submit-task]: rename all references for DISPATCHER_ACCESS_IP to DISPATCHER_ADDRESS

Comment (thiagomanel) [reply bin/submit-task]: also, indicate the user should pass in the format IP:PORT, then remove the constant

File: bin/submit-task (hunk 1 above)
Comment (thiagomanel) [reply bin/submit-task]: ... and, add it as an example in your end-to-end test

File: bin/submit-task [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+INPUTDOWNLOADING_TAG=$9
+PREPROCESSING_TAG=${10}
+PROCESSING_TAG=${11}
+DISPATCHER_ACCESS_IP=${12}
+
+## constants
+DISPATCHER_ACCESS_PORT=8091
+TASK_PRIORITY=0
+
+curl --request POST "http://$DISPATCHER_ACCESS_IP:$DISPATCHER_ACCESS_PORT/processings" --header "Content-Type: text/plain"  --data-raw "userEmail=$USER_EMAIL&userPass=$USER_PASSWORD&lowerLeft[]=$LOWER_LEFT_LATITUDE&lowerLeft[]=$LOWER_LEFT_LONGITUDE&upperRight[]=$UPPER_RIGHT_LATITUDE&upperRight[]=$UPPER_RIGHT_LONGITUDE&initialDate=$PROCESSING_INIT_DATE&finalDate=$PROCESSING_FINAL_DATE&inputGatheringTag=$INPUTDOWNLOADING_TAG&inputPreprocessingTag=$PREPROCESSING_TAG&algorithmExecutionTag=$PROCESSING_TAG&email=$USER_EMAIL&priority=$TASK_PRIORITY"
```
Comment (thiagomanel) [inline bin/submit-task]: couldn't we break some line in the curl command to make it more readable?

**Review by thiagomanel — CHANGES_REQUESTED** [review]
almost done

File: bin/submit-task (hunk 3 above)
Comment (thiagoyeds) [reply bin/submit-task]: Done

File: bin/submit-task (hunk 2 above)
Comment (thiagoyeds) [reply bin/submit-task]: Done

File: bin/submit-task (hunk 1 above)
Comment (thiagoyeds) [reply bin/submit-task]: Done

File: bin/submit-task [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+UPPER_RIGHT_LONGITUDE=$6
+PROCESSING_INIT_DATE=$7
+PROCESSING_FINAL_DATE=$8
+INPUTDOWNLOADING_TAG=$9
+PREPROCESSING_TAG=${10}
+PROCESSING_TAG=${11}
+DISPATCHER_ADDRESS=${12}
+
+## constants
+TASK_PRIORITY=0
```
Comment (wesleymonte) [inline bin/submit-task]: Add `readonly`.

File: bin/submit-task [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+curl --request POST "http://$DISPATCHER_ADDRESS/processings" \
+  --header "Content-Type: text/plain"  \
+  --data-raw "userEmail=$USER_EMAIL&userPass=$USER_PASSWORD&lowerLeft[]=$LOWER_LEFT_LATITUDE&lowerLeft[]=$LOWER_LEFT_LONGITUDE&upperRight[]=$UPPER_RIGHT_LATITUDE&upperRight[]=$UPPER_RIGHT_LONGITUDE&initialDate=$PROCESSING_INIT_DATE&finalDate=$PROCESSING_FINAL_DATE&inputGatheringTag=$INPUTDOWNLOADING_TAG&inputPreprocessingTag=$PREPROCESSING_TAG&algorithmExecutionTag=$PROCESSING_TAG&email=$USER_EMAIL&priority=$TASK_PRIORITY"
```
Comment (wesleymonte) [inline bin/submit-task]: Add braces on variables for disambiguation. e.g. `${DISPATCHER_ADDRESS}`.

**Review by wesleymonte — CHANGES_REQUESTED** [review]

File: bin/submit-task (hunk 5 above)
Comment (thiagoyeds) [reply bin/submit-task]: Done

File: bin/submit-task (hunk 4 above)
Comment (thiagoyeds) [reply bin/submit-task]: Done

File: bin/submit-task (hunk 5 above)
Comment (wesleymonte) [reply bin/submit-task]: Do it for all variables in curl request.

Comment (thiagoyeds) [reply bin/submit-task]: Done

**Review by thiagomanel — APPROVED** [review]
lgtm

**Pull request merged**